src/platform/item/linked-wearable-mappings.ts

```diff
@@ -173,3 +173,103 @@ export namespace Mapping {
+      case MappingType.SINGLE:
+        switch (other.type) {
+          case MappingType.SINGLE:
+            return mapping.id === other.id
+          case MappingType.ANY:
+            return true
+          case MappingType.MULTIPLE:
+            return other.ids.includes(mapping.id)
+          case MappingType.RANGE:
+            return BigInt(mapping.id) >= BigInt(other.from) && BigInt(mapping.id) <= BigInt(other.to)
+        }
+
+      case MappingType.ANY:
+        return true
+
+      case MappingType.MULTIPLE:
+        switch (other.type) {
+          case MappingType.SINGLE:
+            return mapping.ids.includes(other.id)
+          case MappingType.ANY:
+            return true
+          case MappingType.MULTIPLE:
+            return mapping.ids.some((id) => other.ids.includes(id))
+          case MappingType.RANGE:
+            return mapping.ids.some((id) => BigInt(id) >= BigInt(other.from) && BigInt(id) <= BigInt(other.to))
+        }
+
+      case MappingType.RANGE:
+        switch (other.type) {
+          case MappingType.SINGLE:
+            return BigInt(other.id) >= BigInt(mapping.from) && BigInt(other.id) <= BigInt(mapping.to)
+          case MappingType.ANY:
+            return true
+          case MappingType.MULTIPLE:
+            return other.ids.some((id) => BigInt(id) >= BigInt(mapping.from) && BigInt(id) <= BigInt(mapping.to))
+          case MappingType.RANGE:
+            return BigInt(mapping.from) <= BigInt(other.to) && BigInt(mapping.to) >= BigInt(other.from)
+        }
+    }
+  }
+
+  function addMapping(network: Network, contractAddress: ContractAddress, mapping: Mapping) {
+    const lowerContractAddress = contractAddress.toLowerCase()
+    mappings[network] = mappings[network] ?? {}
+    mappings[network]![lowerContractAddress] = mappings[network]![lowerContractAddress] ?? []
+
+    for (const existingMapping of mappings[network]![lowerContractAddress]) {
+      if (overlappingCheck(existingMapping, mapping)) {
+        throw new Error(
+          `Cannot add mapping to contract ${lowerContractAddress} on network ${network} because it overlaps with existing mapping: ${JSON.stringify(
+            existingMapping
+          )}`
+        )
+      }
+    }
+
+    // Safe to add
+    mappings[network]![lowerContractAddress].push(mapping)
+  }
+
+  function includesNft(network: Network, contractAddress: ContractAddress, tokenId: string): boolean {
+    if (!mappings[network]?.[contractAddress]) {
+      return false
+    }
+
+    return mappings[network]![contractAddress].some((mapping) => {
+      switch (mapping.type) {
+        case MappingType.SINGLE:
+          return mapping.id === tokenId
+        case MappingType.ANY:
+          return true
+        case MappingType.MULTIPLE:
+          return mapping.ids.includes(tokenId)
+        case MappingType.RANGE:
+          return BigInt(tokenId) >= BigInt(mapping.from) && BigInt(tokenId) <= BigInt(mapping.to)
+      }
+    })
+  }
+
+  return {
```

Comment:
What do you think about adding a function to get the mapping we've created? It would be really helpful as well.

Reply:
Done